src/e2ee.rs

```diff
@@ -136,30 +141,24 @@ impl EncryptHelper {
         let sign_key = {
             keyring.add_ref(&self.public_key);
             let key = Key::from_self_private(context, self.addr.clone(), &context.sql);
-            if key.is_none() {
-                bail!("no own private key found")
-            }
+            ensure!(key.is_some(), "no own private key found");
+
             key
         };
 
-        /* encrypt message */
+        // encrypt message
         unsafe {
             mailprivacy_prepare_mime(in_out_message);
-            let mut part_to_encrypt: *mut Mailmime =
-                (*in_out_message).mm_data.mm_message.mm_msg_mime;
+            let mut part_to_encrypt = (*in_out_message).mm_data.mm_message.mm_msg_mime;
             (*part_to_encrypt).mm_parent = ptr::null_mut();
-            let imffields_encrypted: *mut mailimf_fields = mailimf_fields_new_empty();
-            /* mailmime_new_message_data() calls mailmime_fields_new_with_version() which would add the unwanted MIME-Version:-header */
-            let message_to_encrypt: *mut Mailmime = mailmime_new(
+            let imffields_encrypted = mailimf_fields_new_empty();
+
+            // mailmime_new_message_data() calls mailmime_fields_new_with_version()
+            // which would add the unwanted MIME-Version:-header
+            let message_to_encrypt = mailmime_new_simple(
```

Comment:
nice, the first fruits of being able to change mmime and dcc-rs at the same time ;)